lr-cli/cortx_setup/commands/storage/config.py

```diff
@@ -290,31 +407,68 @@ def run(self, **kwargs):
+                        self.logger.error("Enclosure ID is not set"
+                            "Run following command to set the enclosure id:"
+                            "cortx_setup storage config --user <user>"
+                            " --password <passwd> --ip <ip> --port <port>")
+                        raise RuntimeError(
+                            "Cannot set mode, ip and port without enclosure id"
                         )
-                        raise RuntimeError("Incomplete arguments provided")
 
                 if user is not None and password is not None:
                     if self.enclosure_id:
-                        host = pillar_get(f"storage/{enc_num}/controller/primary/ip")[node_id][f'storage/{enc_num}/controller/primary/ip']
-                        port = pillar_get(f"storage/{enc_num}/controller/primary/port")[node_id][f'storage/{enc_num}/controller/primary/port']
-
-                        valid_connection_check = EnclosureInfo(host, user, password, port).connection_status()
+                        host = get_pillar_data(
+                            f"storage/{enc_num}/controller/primary/ip"
+                            )[node_id][f'storage/{enc_num}/controller/primary/ip']
+                        port = get_pillar_data(
+                            f"storage/{enc_num}/controller/primary/port"
+                            )[node_id][f'storage/{enc_num}/controller/primary/port']
+
+                        valid_connection_check = EnclosureInfo(
+                                                    host,
+                                                    user,
+                                                    password,
+                                                    port
+                                                ).connection_status()
 
                         if valid_connection_check:
                             self.update_pillar_and_conf('user', user)
                             self.update_pillar_and_conf('password', password)
                         else:
                             self.logger.error(
-                                f"Cannot establish connection with controller using user={user} and password={password} as credentials"
+                                f"Could not establish connection with"
+                                " controller with provided credentials"
                             )
                             raise ValueError("Invalid credentials provided")
                     else:
                         self.logger.error(
-                            "Enclosure ID is not set: Please provide ip and port as well to set Enclosure ID"
+                            "Enclosure ID is not set:"
+                            " Please provide controller ip and port to fetch"
+                            " the enclosure id"
                         )
                         raise RuntimeError("Incomplete arguments provided")
 
+                if cvg > 1:
```

Comment:
CVG count can be any unsigned integer with value greater than 0 and less than number of existing devices...
CVG index will always be from 0 to (CVG count -1)

Reply:
I have removed/updated this condition, please check. Also as mentioned in the prev comment there's no unsigned int in Python.